tests/test_cli_utils.py

```diff
@@ -11,6 +12,235 @@
         ("samples/pipeline_template_examples/testing/example.yaml", False),
     ],
 )
-def test_is_sweep_pipeline(pipeline_file: Path, expected_is_sweep_pipeline: bool):
+def test_is_sweep_pipeline_file(pipeline_file: Path, expected_is_sweep_pipeline: bool):
+    """Test is_sweep_pipeline with file paths"""
     pipeline_file_path = Path(__file__).parent / pipeline_file
     assert is_sweep_pipeline(pipeline_file_path) is expected_is_sweep_pipeline
+
+
+def test_is_sweep_pipeline_dict_with_range():
+    """Test is_sweep_pipeline with a Python dict containing a sweep range"""
+    pipeline_dict = '''[
+        {
+            "method": "normalize",
+            "module_path": "httomolibgpu.prep.normalize",
+            "parameters": {
+                "cutoff": {
+                    "start": 10,
+                    "stop": 15,
+                    "step": 2
+                },
+                "minus_log": true
+            }
+        }
+    ]'''
+    assert is_sweep_pipeline(pipeline_dict) is True
+
+
+def test_is_sweep_pipeline_dict_with_list():
+    """Test is_sweep_pipeline with a Python dict containing a sweep list"""
+    pipeline_dict = '''[
+        {
+            "method": "normalize",
+            "module_path": "httomolibgpu.prep.normalize",
+            "parameters": {
+                "cutoff": [5, 10, 15],
+                "minus_log": true
+            }
+        }
+    ]'''
+    assert is_sweep_pipeline(pipeline_dict) is True
+
+
+def test_is_sweep_pipeline_json_string_with_sweep():
+    """Test is_sweep_pipeline with JSON strings containing both types of sweeps"""
+    # Test with sweep range
+    pipeline_range = '''[
+        {
+            "method": "normalize",
+            "module_path": "httomolibgpu.prep.normalize",
+            "parameters": {
+                "cutoff": {
+                    "start": 10,
+                    "stop": 15,
+                    "step": 2
+                }
+            }
+        }
+    ]'''
+    assert is_sweep_pipeline(pipeline_range) is True
+    
+    # Test with sweep list
+    pipeline_list = '''[
+        {
+            "method": "normalize",
+            "module_path": "httomolibgpu.prep.normalize",
+            "parameters": {
+                "cutoff": [5, 10, 15]
+            }
+        }
+    ]'''
+    assert is_sweep_pipeline(pipeline_list) is True
```

Comment:
Similar question as above, this looks to be pretty much a duplicate of what's in `test_is_sweep_pipeline_dict_with_list`:
```
    pipeline_dict = '''[
        {
            "method": "normalize",
            "module_path": "httomolibgpu.prep.normalize",
            "parameters": {
                "cutoff": [5, 10, 15],
                "minus_log": true
            }
        }
    ]'''
```

If this part of `test_is_sweep_pipeline_json_string_with_sweep` doesn't do anything different to what `test_is_sweep_pipeline_dict_with_list` does, then please remove this, as it's basically a duplicate test.